src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -3459,86 +3458,56 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
                                 out providerInstance);
                 }
 
+                // Don't support 'New-Item -Type Directory' on the Function provider
+                // if the runspace has ever been in constrained language mode, as the mkdir
+                // function can be abused
+                if (context.ExecutionContext.HasRunspaceEverUsedConstrainedLanguageMode &&
```

Comment:
You should add a test case to cover this

Reply:
It is not new code. It is moved code.